### PR TITLE
Fix:BZ1656032:Change the path of openstack cloud provider conf file for static pods

### PR DESCRIPTION
**Pull request by bysnupy**
* Fix: [[DOCS] cloud provider config file must be located into "/etc/origin/cloudprovider"](https://bugzilla.redhat.com/show_bug.cgi?id=1656032)

* Version: `v3.10` and `v3.11+`

* Description:
 As of `v3.10`, the master service (API / Controllers) are provided by [control plane static pods](https://docs.openshift.com/container-platform/3.10/architecture/infrastructure_components/kubernetes_infrastructure.html#control-plane-static-pods), and the pods would be mounted limited directories for config files and certificates and so on. But `/etc` directory is not mounted by static pods.

[https://github.com/openshift/openshift-ansible/blob/release-3.11/roles/openshift_cloud_provider/tasks/openstack.yml#L10]
~~~
- name: Create cloud config from template
  template:
    dest: "{{ openshift.common.config_base }}/cloudprovider/openstack.conf"
    src: openstack.conf.j2
~~~
above `dest` path would be `/etc/origin/cloudprovider`, so it should not be mounted /etc, and only openstack cloud provider pages are used `/etc/cloud.conf` file as example, it's wrong and confusing, refer other cloud provider's pages. (`Google Cloud Engine`, `Azure` ... their path prefix is `/etc/origin/cloudprovider`).  Refer the related BZ for more details. In reality, `/etc/cloud.conf` did not work.

### CODE REVIEW

File: install_config/configuring_openstack.adoc [hunk 1]
```diff
@@ -50,6 +50,12 @@ subnet-id = <UUID of the load balancer subnet>
 Consult your OpenStack administrators for values of the `*OS_*` variables, which
 are commonly used in OpenStack configuration.
 
+[IMPORTANT]
+====
+When triggering a rpm-based installation, only the directories of
```
Comment (kalexand-rh) [inline install_config/configuring_openstack.adoc]:
```
When triggering a rpm-based installation, only the 
*_/etc/origin/cloudprovider_* directory is mounted to the
xref:../architecture/infrastructure_components/kubernetes_infrastructure.adoc#control-plane-static-pods[control plane static pods].
You must place the *_openstack.conf_* file in this folder.
```

File: install_config/configuring_openstack.adoc [hunk 2]
```diff
@@ -120,20 +126,19 @@ kubernetesMasterConfig:
...
+      - "/etc/origin/cloudprovider/openstack.conf"
 ----
 
 [IMPORTANT]
 ====
 When triggering a containerized installation, only the directories of
 *_/etc/origin_* and *_/var/lib/origin_* are mounted to the master and node
-container. Therefore, *_cloud.conf_* should be in *_/etc/origin/_* instead of
-*_/etc/_*.
+container. Therefore, *_openstack.conf_* should be in *_/etc/origin/_*.
```
Comment (kalexand-rh) [inline install_config/configuring_openstack.adoc]: s/should/must

File: install_config/configuring_openstack.adoc [hunk 3]
```diff
@@ -161,8 +166,7 @@ integration will not work.
...
```
Comment (kalexand-rh) [inline install_config/configuring_openstack.adoc]: s/should/must

File: install_config/configuring_openstack.adoc [hunk 4]
```diff
@@ -220,7 +224,7 @@ pvc-1faa6f93-64ac-11e8-930c-fa163e3c373c   1Gi        RWO            Delete
 
 The default setting is enabled. Using the `oc get pv --show-labels` command returns the `failure-domain.beta.kubernetes.io/zone=nova` label.
 
-To disable the zone label, update the cloud.conf file by adding:
+To disable the zone label, update the `openstack.conf` file by adding:
```
Comment (kalexand-rh) [inline install_config/configuring_openstack.adoc]: \*_openstack.conf\_*

**Review by kalexand-rh — COMMENTED** [review]
@bysnupy, I have a few requests. :)

**Comment by kalexand-rh** [comment]
@xltian, will you please suggest a reviewer?

**Comment by bysnupy** [comment]
@kalexand-rh No problem !! Thank you for your kind review and correction :^)
I've changed as your requests and I also squashed the commits. Could you please check it ?

**Comment by xltian** [comment]
@shlao Please review

**Comment by shlao** [comment]
Verified with the OCP on OSP
# grep -A1 cloud-config: /etc/origin/master/master-config.yaml
    cloud-config:
    - /etc/origin/cloudprovider/openstack.conf
--
    cloud-config:
    - /etc/origin/cloudprovider/openstack.conf

File: install_config/configuring_openstack.adoc [hunk 5]
```diff
@@ -120,20 +128,19 @@ kubernetesMasterConfig:
...
+      - "/etc/origin/cloudprovider/openstack.conf"
   controllerArguments:
     cloud-provider:
       - "openstack"
     cloud-config:
-      - "/etc/cloud.conf"
+      - "/etc/origin/cloudprovider/openstack.conf"
 ----
 
 [IMPORTANT]
```
Comment (gpei) [inline install_config/configuring_openstack.adoc]: IMO we don't need this note for 3.10 and 3.11. Master service is only provided via static pod, no rpm-install or containerized install concept for master any more

File: install_config/configuring_openstack.adoc [hunk 6]
```diff
@@ -161,8 +168,7 @@ integration will not work.
 ====
 When triggering a containerized installation, only the directories of
```
Comment (gpei) [inline install_config/configuring_openstack.adoc]: So is this note, `containerized installation` is a deprecated concept. No matter rpm-installed node or system-container installed node, the cloud config file should always be `/etc/origin/cloudprovider/{{ openshift_node_group_cloud_provider }}.conf`

File: install_config/configuring_openstack.adoc [hunk 7]
```diff
@@ -50,6 +50,14 @@ subnet-id = <UUID of the load balancer subnet>
 Consult your OpenStack administrators for values of the `*OS_*` variables, which
 are commonly used in OpenStack configuration.
 
+[IMPORTANT]
+====
+When triggering a rpm-based installation, only the
```
Comment (kalexand-rh) [inline install_config/configuring_openstack.adoc]: @gpei, if we s/When triggering a rpm-based installation/During installation
and remove the two notes about the containerized installation, will that be correct and address all your concerns?

Comment (gpei) [reply install_config/configuring_openstack.adoc]: Yeah, and also please address the directory in `Therefore, *_openstack.conf_* must be in *_/etc/origin/_*.` , which should be _/etc/origin/cloudprovider/_

Comment (kalexand-rh) [reply install_config/configuring_openstack.adoc]: @bysnupy, will you please update this note and rebase?

Comment (bysnupy) [reply install_config/configuring_openstack.adoc]: @kalexand-rh , I've completed the rebase of the codes.

**Comment by bysnupy** [comment]
@kalexand-rh I'm so sorry. This PR has closed by my mistake to remove the related branch. It has caused my poor operation about `git`. I've opened the same `PR` which squashed the commits here.
Refer the following `PR` for details of modified the files.

* `PR`: https://github.com/openshift/openshift-docs/pull/13232

**Pull request closed without merging**